### PR TITLE
CLOUDP-82836: Parse service version

### PR DESCRIPTION
**Pull request by fmenezes**
## Description

Parse service version header `X-MongoDB-Service-Version`

Link to any related issue(s): [CLOUDP-82836](https://jira.mongodb.org/browse/CLOUDP-82836)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: mongodbatlas/mongodbatlas.go [hunk 1]
```diff
@@ -564,3 +566,38 @@ func setListOptions(s string, opt interface{}) (string, error) {
...
+			result.GitHash = keyvalue[1]
+		case "versionString":
+			result.Version = keyvalue[1]
+		}
+	}
+	return &result
+}
+
+// GetServiceVersion parses version information returned in the response.
+func (resp *Response) GetServiceVersion() *ServiceVersion {
```
Comment (gssbzn) [inline mongodbatlas/mongodbatlas.go:601]: [nit] getters are non idiomatic on go
```suggestion
func (resp *Response) ServiceVersion() *ServiceVersion {
```

File: mongodbatlas/service_version.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+package mongodbatlas
+
+import (
+	"context"
+	"net/http"
+)
+
+const versionPath = "api/private/unauth/version"
+
+// ServiceVersionService is an interface for the version private endpoint of the MongoDB Atlas API.
```
Comment (gssbzn) [inline mongodbatlas/service_version.go:24]: we tend to leave a note about this not being supported outside what can be seen here

https://github.com/mongodb/go-client-mongodb-atlas/blob/c271f9850cb0408445a31f2df0a920687c775524/mongodbatlas/ip_info.go#L26-L27

File: mongodbatlas/mongodbatlas.go (hunk 1 above)
Comment (fmenezes) [reply mongodbatlas/mongodbatlas.go:601]: I know it does not sound right but I did not think `Get` as a getter here 😄

File: mongodbatlas/mongodbatlas.go [hunk 3]
```diff
@@ -564,3 +566,38 @@ func setListOptions(s string, opt interface{}) (string, error) {
...
+	var result ServiceVersion
+	pairs := strings.Split(s, ";")
+	for _, pair := range pairs {
+		keyvalue := strings.Split(strings.TrimSpace(pair), "=")
+		switch keyvalue[0] {
+		case "gitHash":
+			result.GitHash = keyvalue[1]
+		case "versionString":
+			result.Version = keyvalue[1]
+		}
```
Comment (colm-quinn) [inline mongodbatlas/mongodbatlas.go:595]: [q] Should we log/fail on any other case? I generally don't like switches without a default, as changes can go undetected. If we had a default error, along with a unit test, a structure change would immediately fail here.

**Review by gssbzn — APPROVED** [review]
LGTM thanks for those changes

**Pull request merged**